assets/src/edit-story/app/uploader/useUploader.js

```diff
@@ -82,41 +82,41 @@ function useUploader() {
         throw createError('PermissionError', file.name, message);
       }
 
-      // The file is too large for the site anyway, abort.
-      if (!isFileSizeWithinLimits(file)) {
-        const message = sprintf(
-          /* translators: first %s is the file size in MB and second %s is the upload file limit in MB */
-          __(
-            'Your file is %1$sMB and the upload limit is %2$sMB. Please resize and try again!',
-            'web-stories'
-          ),
-          bytesToMB(file.size),
-          bytesToMB(maxUpload)
-        );
-        throw createError('SizeError', file.name, message);
-      }
-
-      // TODO: Move this check to useUploadMedia?
-      if (!isValidType(file) && !canTranscodeFile) {
-        let message = __(
-          'No file types are currently supported.',
-          'web-stories'
-        );
-
-        if (allowedFileTypes.length) {
-          /* translators: %s is a list of allowed file extensions. */
-          message = sprintf(
-            /* translators: %s: list of allowed file types. */
-            __('Please choose only %s to upload.', 'web-stories'),
-            translateToExclusiveList(allowedFileTypes)
+      if (!canTranscodeFile) {
+        // The file is too large for the site anyway, abort.
+        if (!isFileSizeWithinLimits(file)) {
+          const message = sprintf(
+            /* translators: first %s is the file size in MB and second %s is the upload file limit in MB */
+            __(
+              'Your file is %1$sMB and the upload limit is %2$sMB. Please resize and try again!',
+              'web-stories'
+            ),
+            bytesToMB(file.size),
+            bytesToMB(maxUpload)
           );
+          throw createError('SizeError', file.name, message);
         }
 
-        throw createError('ValidError', file.name, message);
-      }
+        // TODO: Move this check to useUploadMedia?
+        if (!isValidType(file)) {
+          let message = __(
+            'No file types are currently supported.',
+            'web-stories'
+          );
 
-      // TODO: Move this check to useUploadMedia?
-      if (isFileTooLarge) {
+          if (allowedFileTypes.length) {
+            /* translators: %s is a list of allowed file extensions. */
+            message = sprintf(
+              /* translators: %s: list of allowed file types. */
+              __('Please choose only %s to upload.', 'web-stories'),
+              translateToExclusiveList(allowedFileTypes)
+            );
+          }
+
+          throw createError('ValidError', file.name, message);
+        }
+        // TODO: Move this check to useUploadMedia?
+      } else if (isFileTooLarge) {
```

Comment:
I don't think `else if` is correct here.

This `isFileTooLarge` check is for `MEDIA_TRANSCODING_MAX_FILE_SIZE`, i.e. to see if the file is too large for transcoding.

Let's add some more unit tests for `useUploader` so we know there's no regression.

Reply:
> I don't think else if is correct here.

Why not?

To be honest, I reviewed the existing tests and they seemed pretty solid. What kind of tests should add?

Reply:
This introduction of new code without breaking any existing tests proves that there's no test coverage for this newly changed logic.

Reply:
> This introduction of new code without breaking any existing tests proves that there's no test coverage for this newly changed logic.

Maybe. Either way I have added some new tests.